toolchain/sem_ir/typed_insts.h

```diff
@@ -868,6 +868,25 @@ struct ImplDecl {
   DeclInstBlockId decl_block_id;
 };
 
+// A symbolic witness that a type implements an interface. It represented a
+// promise that the query can be satisfied without providing which impl
```

Comment:
I'm maybe short on context, but I don't feel confident about what's being talked about which kind of query is meant here. Maybe:

```suggestion
// promise that an impl lookup query can be satisfied without providing which impl
```

Reply:
Done